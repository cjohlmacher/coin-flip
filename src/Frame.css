.Frame-Button {
    border-style: none;
    border: solid 1px rgb(255,255,255,0);
    border-radius: 10px;
    background-color: rgba(207, 221, 9, 0.5);
    /*box-shadow effect by Frederico Dossena */
    box-shadow: inset 0 -0.6em 1em -0.35em rgba(0,0,0,0.17),inset 0 0.6em 2em -0.3em rgba(255,255,255,0.15),inset 0 0 0em 0.05em rgba(255,255,255,0.12);
}

.Frame-Button:hover {
    border: solid 1px black;
}