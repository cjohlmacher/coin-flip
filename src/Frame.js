import React from 'react';
import './Frame.css';
import Coin from './Coin';
import Banner from './Banner';
import InfoBar from './InfoBar';
import {useState} from 'react';

const Frame = ({title}) => {
    const [isHeads,setHeads] = useState(true);
    const [count,setCount] = useState({heads: 0, tails: 0});

    const flipCoin = () => {
        const val = Math.random();
        if (val >= 0.5) { 
            setHeads(true);
            setCount(
                {
                    ...count,
                    heads:count.heads+1
                }
            );
        } else {
            setHeads(false);
            setCount(
                {
                    ...count,
                    tails:count.tails+1
                }
            );
        };
    };

    return (
        <div className="Frame">
            <Banner title={title}/>
            <Coin isHeads={isHeads}/>
            <button className="Frame-Button" onClick={flipCoin}>Flip the coin!</button>
            <InfoBar count={count}/>
        </div>
    )
};

Frame.defaultProps = {
    title: 'Coin Flip App',
}

export default Frame;