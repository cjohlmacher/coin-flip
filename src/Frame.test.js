import { fireEvent, render } from '@testing-library/react';
import Frame from './Frame';

beforeEach(function() {
    jest
        .spyOn(Math, "random")
        .mockReturnValueOnce(0.25)
        .mockReturnValueOnce(0.75);
});
  
afterEach(function() {
    Math.random.mockRestore();
});


it('renders without crashing', () => {
    render(<Frame />);
});

it('matches snapshot', () => {
    const {asFragment} = render(<Frame />);
    expect(asFragment()).toMatchSnapshot();
});

it('changes image on coin side flip', () => {
    const { getByAltText, getByRole } = render(<Frame />);
    const image = getByAltText('coin-heads');
    const flipButton = getByRole('button',{name: 'Flip the coin!'});
    fireEvent.click(flipButton);
    expect(image).toHaveAttribute('alt','coin-tails');
    fireEvent.click(flipButton);
    expect(image).toHaveAttribute('alt','coin-heads');
});

it('increments count on flip', () => {
    const { getByRole, getByText } = render(<Frame />);
    const flipButton = getByRole('button',{name: 'Flip the coin!'});
    const headsCount = getByText('Heads: 0');
    const tailsCount = getByText('Tails: 0');
    fireEvent.click(flipButton);
    expect(tailsCount).toHaveTextContent('Tails: 1');
    fireEvent.click(flipButton);
    expect(headsCount).toHaveTextContent('Heads: 1');
});