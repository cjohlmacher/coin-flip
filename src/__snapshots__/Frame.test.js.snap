// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Frame"
  >
    <div
      class="Banner"
    >
      <h1>
        Coin Flip App
      </h1>
    </div>
    <div
      class="Coin"
    >
      <img
        alt="coin-heads"
        class="Coin-Image"
        src="https://images.pexels.com/photos/7702774/pexels-photo-7702774.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
      />
    </div>
    <button
      class="Frame-Button"
    >
      Flip the coin!
    </button>
    <div
      class="InfoBar"
    >
      <p>
        Heads: 0
      </p>
      <p>
        Tails: 0
      </p>
    </div>
  </div>
</DocumentFragment>
`;
